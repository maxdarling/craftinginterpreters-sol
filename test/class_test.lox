{
class DevonshireCream {
  serveOn() {
    return "Scones";
  }
}

print DevonshireCream; // Prints "DevonshireCream".
}

class DevonshireCream {
  serveOn() {
    return "Scones";
  }
}

print DevonshireCream; // Prints "DevonshireCream".

class Bagel {}
var bagel = Bagel();
print bagel; // Prints "Bagel instance".

class Bacon {
  eat() {
    print "Crunch crunch crunch!";
  }
}

Bacon().eat(); // Prints "Crunch crunch crunch!".

class Cake {
  taste() {
    var adjective = "delicious";
    print "The " + this.flavor + " cake is " + adjective + "!";
  }
}

var cake = Cake();
cake.flavor = "German chocolate";
cake.taste(); // Prints "The German chocolate cake is delicious!".

// print this;

// class Foo {
//   init() {
//     return "something else";
//   }
// }

class Foo {
  init() {
    print this;
  }
}

var foo = Foo();
print foo.init();

{
class Foo {
  init() {
    return;
  }
}
print Foo();
}