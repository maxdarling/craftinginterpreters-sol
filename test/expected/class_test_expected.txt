DevonshireCream
DevonshireCream
Bagel instance
Crunch crunch crunch!
The German chocolate cake is delicious!
Foo instance
Foo instance
Foo instance
Foo instance
